test-utils/waitFor.ts

```diff
@@ -0,0 +1,8 @@
+export function waitFor(conditionFunction: () => boolean) {
+  const poll = (resolve: (value: unknown) => void) => {
+    if (conditionFunction()) resolve(null);
+    else setTimeout(() => poll(resolve), 100);
+  };
```

Comment:
At some point after x iterations should we reject? Perhaps a `failAfter` param?

Reply:
done